Source/WebCore/dom/Document.cpp

```diff
@@ -2920,6 +2923,10 @@ void Document::resolveStyle(ResolveStyleType type)
 
     if (CheckedPtr styleOriginatedTimelinesController = this->styleOriginatedTimelinesController())
         styleOriginatedTimelinesController->documentDidResolveStyle();
+
+    // Re-evaluate speculation rules after style changes to ensure CSS selector
+    // matching works with updated styles.
+    considerSpeculationRules();
```

Comment:
I don't think I understand how style resolution could change speculation rules.

Reply:
Since speculation rules can operate on CSS selectors, DOM changes can change the elements on which the rules are applied. See https://searchfox.org/wubkat/source/LayoutTests/imported/w3c/web-platform-tests/speculation-rules/prefetch/document-rules.https.html#154

`resolveStyle()` is a hook that reacts to such DOM changes. I'll fix up the comment to make it more correct and to make that clearer.